packages/cli/src/cmds/init/initHandler.ts

```diff
@@ -30,133 +32,77 @@ import {
-    }, {} as YargsArgs);
+  const isVersionSupported = satisfiesVersionRange(
+    process.version,
+    SATISFIED_VERSION,
+  );
 
-  const isTypeScript = isTemplateTypescript(argv.template as TemplateType);
-  try {
-    await fs.writeFile(
-      NpmSnapFileNames.Manifest,
-      `${JSON.stringify(snapManifest, null, 2)}\n`,
-    );
-  } catch (err) {
+  if (!isVersionSupported) {
     logError(
-      `Init Error: Failed to write '${NpmSnapFileNames.Manifest}'.`,
-      err,
+      `Init Error: You are using an outdated version of Node (${process.version}). Please update to Node ${SATISFIED_VERSION}.`,
     );
-    throw err;
+    throw new Error('outdated node version');
   }
 
-  console.log(`\nInit: Created '${NpmSnapFileNames.Manifest}'.`);
-
-  // Write main .js entry file
-  const { src } = newArgs;
-
-  try {
-    if (pathUtils.basename(src) !== src) {
-      await mkdirp(pathUtils.dirname(src));
-    }
-
-    await fs.writeFile(
-      src,
-      isTypeScript ? template.typescriptSource : template.source,
+  const gitExists = isGitInstalled();
+  if (!gitExists) {
+    logError(
+      `Init Error: git is not installed. Please intall git to continue.`,
     );
-
-    console.log(`Init: Created '${src}'.`);
-  } catch (err) {
-    logError(`Init Error: Failed to write '${src}'.`, err);
-    throw err;
+    throw new Error('git is not installed');
   }
 
-  // Write index.html
-  try {
-    await fs.writeFile(
-      'index.html',
-      isTypeScript ? template.typescriptHtml : template.html,
-    );
+  const directoryToUse = directory
+    ? pathUtils.join(process.cwd(), directory)
+    : process.cwd();
 
-    console.log(`Init: Created 'index.html'.`);
-  } catch (err) {
-    logError(`Init Error: Failed to write 'index.html'.`, err);
-    throw err;
-  }
+  console.log(`Preparing ${directoryToUse} ...`);
 
-  // Write tsconfig.json
-  if (isTypeScript) {
-    try {
-      await fs.writeFile('tsconfig.json', template.typescriptConfig);
-      console.log(`Init: Created 'tsconfig.json'.`);
-    } catch (err) {
-      logError(`Init Error: Failed to write 'tsconfig.json'.`, err);
-      throw err;
-    }
-  }
+  await prepareWorkingDirectory(directoryToUse);
 
-  // Write config file
   try {
-    const defaultConfig: SnapConfig = {
-      cliOptions: newArgs,
-    };
-    const defaultConfigFile = `module.exports = ${JSON.stringify(
-      defaultConfig,
-      null,
-      2,
-    )}
-    `;
-    await fs.writeFile(CONFIG_FILE, defaultConfigFile);
-    console.log(`Init: Wrote '${CONFIG_FILE}' config file`);
+    console.log(`Cloning template...`);
+    await cloneTemplate(directoryToUse);
+
+    fs.rm(pathUtils.join(directoryToUse, '.git'), {
+      force: true,
+      recursive: true,
+    });
   } catch (err) {
-    logError(`Init Error: Failed to write '${CONFIG_FILE}'.`, err);
+    logError('Init Error: Failed to create template, cleaning...');
+    fs.rm(directoryToUse, {
+      force: true,
+      recursive: true,
+    });
```

Comment:
Is this necessary? It could potentially be destructive, if there's some kind of bug in our code. We don't want to wipe random folders in the user's file system. 😅

Reply:
No this is not necessary, I just wanted to clean some of the mess it creates if it fails but you're right that it could be destructive.